Author: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
Date:   Mon Nov 6 22:39:31 2023 +0000

    wifi: mt76: mt7915: also MT7981 is 3T3R but nss2 on 5 GHz band
    
    Just like MT7916 also MT7981 can handle 3T3R DBDC frontend and should
    hence be included in the corresponding conditional expression in the
    driver. Add it.
    
    Fixes: 6bad146d162e ("wifi: mt76: mt7915: add support for MT7981")
    Signed-off-by: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
    Signed-off-by: Daniel Golle <daniel@makrotopia.org>
    Signed-off-by: Felix Fietkau <nbd@nbd.name>
---

--- a/mt7915/main.c
+++ b/mt7915/main.c
@@ -1059,8 +1059,9 @@
 
 	phy->mt76->antenna_mask = tx_ant;
 
-	/* handle a variant of mt7916 which has 3T3R but nss2 on 5 GHz band */
-	if (is_mt7916(&dev->mt76) && band && hweight8(tx_ant) == max_nss)
+	/* handle a variant of mt7916/mt7981 which has 3T3R but nss2 on 5 GHz band */
+	if ((is_mt7916(&dev->mt76) || is_mt7981(&dev->mt76)) &&
+	    band && hweight8(tx_ant) == max_nss)
 		phy->mt76->chainmask = (dev->chainmask >> chainshift) << chainshift;
 	else
 		phy->mt76->chainmask = tx_ant << (chainshift * band);
